File: RatTrack/app/src/main/java/getwreckt/cs2340/rattrack/model/Borough.java
package getwreckt.cs2340.rattrack.model;

/**
 * Created by maya v on 10/20/2017.
 */

public enum Borough {
    BRONX ("Bronx"),
    BROOKLYN ("Brooklyn"),
    MANHATTAN ("Manhattan"),
    QUEENS ("Queens"),
    STATEN_ISLAND ("Staten Island"),
    UNKNOWN ("Unknown"),
    NA ("N/A"),
    NULL ("");

    private String name;

    Borough(String name) {
        this.name = name;
    }

    /**
     * Getter method for name
     * @return name of an admin in string
     */
    public String getName() {
        return this.name;
    }

    /**
     * Setter method for name
     * @param name name of an admin
     */
    public void setName(String name) {
        this.name = name;
    }

    /**
     * Gets Borough in UpperCase
     * @param name name of borough
     * @return borough in UpperCase or null
     */
    public static Borough get(String name) {
        for (Borough borough : values()) {
            if (borough.getName().toUpperCase().equals(name.toUpperCase())) {
                return borough;
            }
        }

        return NULL;
    }
}

File: RatTrackiOS/RatTrackiOS/Model/Borough.swift
//
//  Borough.swift
//  RatTrackiOS
//
//  Created by Mihir Parshionikar on 12/2/17.
//  Copyright © 2017 getwreckt. All rights reserved.
//

import Foundation

//NOTE: this will require a different implementation from how we used this in Java.  I did not know how to convert the java code into swift.

enum BoroughEnum {
    case BRONX
    case BROOKLYN
    case MANHATTAN
    case QUEENS
    case STATEN_ISLAND
    case UNKNOWN
    case NA
    case NULL
}
//An object that holds the enum
class Borough {
    var bor : BoroughEnum
    init(name: String) {
        if (name == "Bronx") {
            bor = BoroughEnum.BRONX
        } else if (name == "Brooklyn") {
            bor = BoroughEnum.BROOKLYN
        } else if (name == "Manhattan") {
            bor = BoroughEnum.MANHATTAN
        } else if (name == "Queens") {
            bor = BoroughEnum.QUEENS
        } else if (name == "Staten Island") {
            bor = BoroughEnum.STATEN_ISLAND
        } else if (name == "Unknown") {
            bor = BoroughEnum.UNKNOWN
        } else if (name == "N/A") {
            bor = BoroughEnum.NA
        } else {
            bor = BoroughEnum.NULL
        }
    }
    
    func get() -> BoroughEnum {
        return bor
    }
}
